### PR TITLE
Remove uneeded Statement::closeCursor() calls.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Database/Schema/SqliteSchemaDialect.php [hunk 1]
```diff
@@ -294,7 +294,6 @@ public function convertIndexDescription(TableSchema $schema, array $row): void
         foreach ($statement->fetchAll('assoc') as $column) {
             $columns[] = $column['name'];
         }
-        $statement->closeCursor();
```
Comment (othercorey) [inline src/Database/Schema/SqliteSchemaDialect.php:297]: We could clean up code like above with `array_column()`.

File: src/ORM/Table.php [hunk 2]
```diff
@@ -1731,7 +1731,6 @@ public function updateAll(
             ->set($fields)
             ->where($conditions)
             ->execute();
-        $statement->closeCursor();
```
Comment (othercorey) [inline src/ORM/Table.php:1731]: I assume the RETURNING clause doesn't require this, but I don't know for sure.

File: src/Database/Schema/SqliteSchemaDialect.php (hunk 1 above)
Comment (ADmad) [reply src/Database/Schema/SqliteSchemaDialect.php:297]: Can be done in a follow up PR.

File: src/ORM/Table.php (hunk 2 above)
Comment (ADmad) [reply src/ORM/Table.php:1731]: We can always add back the `closeCursor()` if we start having issues with inserts, which use `RETURNING` :slightly_smiling_face:.

**Comment by othercorey** [comment]
Looks ok to me.

**Comment by markstory** [comment]
I thought we closed cursor *after* fetching the rows to avoid problems with streaming queries. Perhaps this is another win from only offering buffered queries and statements now.

**Comment by othercorey** [comment]
> I thought we closed cursor _after_ fetching the rows to avoid problems with streaming queries. Perhaps this is another win from only offering buffered queries and statements now.

It should be necessary only when there are unfetched rows. It basically frees up the statement to run again, but if your DB doesn't support parallel fetching, it can block the queries. It seemed like we were trying to avoid the 2nd issue with most of the code.

However, if we've fetched the rows, then the connection should be feed up automatically.

https://www.php.net/manual/en/pdostatement.closecursor.php

**Comment by othercorey** [comment]
Do we even have a DB that doesn't support parallel SELECT on the same connection?

**Comment by ADmad** [comment]
> I thought we closed cursor after fetching the rows to avoid problems with streaming queries.

Based on what I have read on various sources, closing the cursor is only necessary if you don't fetch *all* possible rows for a statement. So it shouldn't be necessary at all for non SELECT statement and for SELECTs only when you do a `fetch()` and then are still more rows available.

**Comment by markstory** [comment]
> Do we even have a DB that doesn't support parallel SELECT on the same connection?

Not currently we don't.

**Comment by ADmad** [comment]
So can this be merged?

**Pull request merged**